sample/src/main/kotlin/com/slack/circuit/sample/petdetail/PetDetail.kt

```diff
@@ -121,35 +124,49 @@ class PetDetailScreenFactory @Inject constructor() : ScreenViewFactory {
   }
 }
 
-private fun petDetailUi() = ui<PetDetailScreen.State, Nothing> { state, _ -> RenderImpl(state) }
+private fun petDetailUi() = ui<PetDetailScreen.State, Nothing> { state, _ -> PetDetail(state) }
 
 @Composable
-private fun RenderImpl(state: PetDetailScreen.State) {
+internal fun PetDetail(state: PetDetailScreen.State) {
   Scaffold(modifier = Modifier.systemBarsPadding()) { padding ->
     when (state) {
       PetDetailScreen.State.Loading -> {
         Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
-          CircularProgressIndicator()
+          CircularProgressIndicator(modifier = Modifier.testTag("progress"))
+        }
+      }
+      PetDetailScreen.State.UnknownAnimal -> {
+        Box(modifier = Modifier.fillMaxSize(), contentAlignment = Alignment.Center) {
+          Text(
+            modifier = Modifier.testTag("unknown animal"),
+            text = stringResource(id = R.string.unknown_animals)
+          )
         }
       }
-      PetDetailScreen.State.NoAnimal -> TODO()
       is PetDetailScreen.State.Success -> {
-        LazyColumn(modifier = Modifier.padding(padding)) {
+        LazyColumn(modifier = Modifier.padding(padding).testTag("animal container")) {
           item {
             AsyncImage(
-              modifier = Modifier.fillMaxWidth(),
+              modifier = Modifier.fillMaxWidth().testTag("image"),
               model =
                 ImageRequest.Builder(LocalContext.current)
                   .data(state.photoUrl)
+                  .fallback(R.drawable.dog)
                   .placeholderMemoryCacheKey(state.photoUrlMemoryCacheKey)
                   .crossfade(true)
                   .build(),
               contentDescription = state.name,
               contentScale = ContentScale.FillWidth,
             )
           }
-          item { Text(text = state.name, style = MaterialTheme.typography.displayLarge) }
-          item { Text(text = state.description) }
+          item {
+            Text(
+              modifier = Modifier.testTag("name"),
```

Comment:
nit: I think we can just assert the text content instead of looking for the specific test tag? Otherwise same here with the shared const for the tag.

Reply:
Yup, good point. I left a few tags in place as they simplified testing. For most cases, I grabbed nodes by asserting the expected text. So `onNodeWithText` instead of `onNodeWithTag`